derive-macros/src/util.rs

```diff
@@ -0,0 +1,64 @@
+use proc_macro2::{Span, TokenStream};
+use syn::parse::{Parse, ParseStream};
+
+/// A span and a value.
+///
+/// Mainly used to keep spans around for generating errors later on.
+#[derive(Debug)]
+pub struct WithSpan<T> {
```

Comment:
This is a new concept to me and I would like to clarify is the reason to have a WithSpan. Is it because the span is lost somewhere in the process of parsing the tokens through Syn?

Reply:
Not if you keep a `LitInt` around for example. But if you parse it further to a `i32` you would loose the span. I don't want to keep parsing the `LitInt` everything I want to know the value, so I added this struct to keep the parsed result but also the span.